src/admin/index.js

```diff
@@ -595,6 +595,44 @@ module.exports = ({
     }
   }
 
+  /**
+   * List groups in a broker
+   *
+   * @return {Promise<ListGroups>}
+   *
+   * @typedef {Object} ListGroups
+   * @property {Array<ListGroup>} groups
+   *
+   * @typedef {Object} ListGroup
+   * @property {string} groupId
+   * @property {string} protocolType
+   */
+  const listGroups = async () => {
+    const retrier = createRetry(retry)
+
+    return retrier(async (bail, retryCount, retryTime) => {
+      try {
+        await cluster.refreshMetadata()
+
+        let groups = []
+        for (var nodeId in cluster.brokerPool.brokers) {
+          const broker = await cluster.findBroker({ nodeId })
+          const response = await broker.listGroups()
+          groups = groups.concat(response.groups)
+        }
+
+        return { groups }
+      } catch (e) {
+        if (e.type === 'NOT_CONTROLLER') {
```

Comment:
You don't need this catch, since you are querying all brokers, not just the controller. `cluster.findBroker` has internal retries that handle things like outdated metadata already, so that should be fine.

Reply:
Solved, removed also fetching metadata as it is not needed for this case

Reply:
Are you sure that's not needed? Because we need metadata to get the list of brokers, and the cached metadata might be outdated.

Reply:
True, returned it